x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/RuntimeKeywordMappedFieldType.java

```diff
@@ -62,13 +63,27 @@ public Object valueForDisplay(Object value) {
 
     @Override
     public String typeName() {
-        // TODO not sure what we should return here: the runtime type or the field type?
-        // why is the same string returned from three different methods?
         return ScriptFieldMapper.CONTENT_TYPE;
     }
 
+    @Override
+    public String familyTypeName() {
+        return KeywordFieldMapper.CONTENT_TYPE;
+    }
+
+    @Override
+    public boolean isSearchable() {
+        return true;
+    }
+
+    @Override
+    public boolean isAggregatable() {
+        return true;
+    }
+
     @Override
     public ScriptBinaryFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
+        assert fullyQualifiedIndexName.length() > 0 : "index name must not be empty";
```

Comment:
I think that I will remove this assertion. I initially assumed that calling this would make us load doc_values for a runtime field which is expensive, but it will only make us instantiate the classes and doc_values will not be advanced, hence no script is going to be executed.

Reply:
:+1: